src/lib.rs

```diff
@@ -7,10 +7,29 @@
 use log::*;
 
 use std::collections::HashMap;
-use std::fs::File;
+use std::fs::{self, File};
 use std::io::{BufRead, BufReader, Read, Write};
 use std::path::{Path, PathBuf};
 
+macro_rules! update_and_test {
+    ($self: ident, $set_func:ident, $value:expr, $get_func:ident) => {
+        if $value.is_some() {
+            let _ = $self.$set_func($value.unwrap());
```

Comment:
How about use `$self.$set_func($value.unwrap());` to let it return if error occurred.

Reply:
done